torchaudio/datasets/tedlium.py

```diff
@@ -0,0 +1,182 @@
+                )
+            )
+
+        basename = os.path.basename(url)
+        archive = os.path.join(root, basename)
+
+        basename = basename.split(".")[0]
+
+        self._path = os.path.join(root, folder_in_archive, _RELEASE_CONFIGS[release]["data_path"])
+        if subset in ["train", "dev", "test"]:
+            self._path = os.path.join(self._path, subset)
+        if download:
+            if not os.path.isdir(self._path):
+                if not os.path.isfile(archive):
+                    checksum = _RELEASE_CONFIGS[release]["checksum"]
+                    download_url(url, root, hash_value=checksum)
+                extract_archive(archive)
+
+        self._walker = []
+
+        # Create walker for all samples
+        for file in walk_files(self._path, suffix=".stm", prefix=False, remove_suffix=True):
+            stm_path = os.path.join(self._path, "stm", file + ".stm")
+            with open(stm_path) as f:
+                l = len(f.readlines())
+                self._walker.extend((file, line) for line in range(l))
+        # Read phoneme dictionary
+        dict_path = os.path.join(root, folder_in_archive, _RELEASE_CONFIGS[release]["dict"])
+        self.phoneme_dict = {}
+        with open(dict_path, "rb") as f:
+            for line in f.readlines():
+                content = line.decode("utf-8").strip("\n").split(" ", 1)
+                if len(content) > 1:
+                    key, value = content[0], content[1]
+                    self.phoneme_dict[key] = value.strip()
+                    # # Some lines in release1 dont have a phoneme for a word so value will be out of index
+                    # # Need to find a better solution to read the dictionary
+
+    def load_tedlium_item(self, fileid: str, line: int, path: str) -> Tedlium_item:
+        """Loads a TEDLIUM dataset sample given a file name and corresponding sentence name
+
+        Args:
+            fileid (str): File id to identify both text and audio files corresponding to the sample
+            line (int): Line identifier for the sample inside the text file
+            path (str): Dataset root path
+
+        Returns:
+            Tedlium_item: A namedTuple containing [waveform, sample_rate, transcript, talk_id, speaker_id, identifier]
+        """
+        transcript_path = os.path.join(path, "stm", fileid)
+        with open(transcript_path + ".stm") as f:
+            transcript = f.readlines()[line]
+            talk_id, _, speaker_id, start_time, end_time, identifier, transcript = transcript.split(" ", 6)
+
+        wave_path = os.path.join(path, "sph", fileid)
+        waveform, sample_rate = self.__load_audio__(
+            wave_path + self._ext_audio, start_time=start_time, end_time=end_time
+        )
+
+        return Tedlium_item(waveform, sample_rate, transcript, talk_id, speaker_id, identifier)
+
+    def __load_audio__(self, path: str, start_time: float, end_time: float, sample_rate: int = 16000) -> [Tensor, int]:
```

Comment:
double underscore prefix and suffix are reserved for specific purposes in Python. Please use single underscore prefix (and no suffix)

> \_\_double_leading_and_trailing_underscore__: "magic" objects or attributes that live in user-controlled namespaces. E.g. \_\_init__, \_\_import__ or \_\_file__. Never invent such names; only use them as documented.

Reference: [PEP8](https://www.python.org/dev/peps/pep-0008/)
This page summarizes the naming convention for function and variable https://dbader.org/blog/meaning-of-underscores-in-python

Reply:
Oh thanks, I didnt knew!